### ISSUE TITLE
IllegalArgumentException: "At least one shrinkable is required" during shrinking hides cause of test failure

### ISSUE THREAD

**Issue by daniel-huss** [jqwik-team/jqwik#522]
## Testing Problem

Minimal(ish?) reproducer:

```java

import net.jqwik.api.*;
import net.jqwik.api.domains.Domain;
import net.jqwik.api.domains.DomainContextBase;

import java.util.stream.Collectors;
import java.util.stream.IntStream;

@Domain(ExceptionDuringShrinkingTest.MyDomain.class)
class ExceptionDuringShrinkingTest {
    @Property
    void test(@ForAll NotRelevant obj) {
        throw new RuntimeException("Hello!");
    }

    public static class NotRelevant {
    }

    public static class MyDomain extends DomainContextBase {
        @Provide
        public static Arbitrary<NotRelevant> objects() {
            return Arbitraries.subsetOf(IntStream.range(0, 1000).boxed().collect(Collectors.toSet()))
                              .flatMap(subset -> Combinators
                                      .combine(subset.stream().map(Arbitraries::just).collect(Collectors.toList()))
                                      .as(ignored -> new NotRelevant())
                              );
        }
    }
}
```

When this test fails, I get
 ```
timestamp = 2023-09-11T09:50:23.088071, ExceptionDuringShrinkingTest:test = 
  java.util.concurrent.ExecutionException:
    java.lang.IllegalArgumentException: At least one shrinkable is required
```

but the RuntimeException that failed the test is not printed. To my best knowledge the above usage of flatMap / combine is technically valid - apologies if it's not.

Found using version 1.8.0.

**Comment by jlink** [comment]
@daniel-huss Thanks for creating the issue. Looks like a bug at first glance. I'll check later this week.

**Comment by jlink** [comment]
@daniel-huss Sorry for the delay. It's a bug. Working on it right now.

**Comment by jlink** [comment]
Fixed and released in 1.8.0-SNAPSHOT

**Comment by jlink** [comment]
1.8.1 has been released.
